### PR TITLE
ValidatedFilePath

### PR DESCRIPTION
**Pull request by Ajeett01**
## Related Issue
- [bug]: Uncontrolled data used in path expression
Closes: #1558 

#### Describe the changes you've made
Incorporating the path validation step before performing file operations.

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Please let us know if any test cases are added
NA

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
NA

## Checklist:
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

### CODE REVIEW

File: pkg/platform/yaml/testReport.go [hunk 1]
```diff
@@ -97,9 +97,14 @@ func (fe *TestReport) Write(ctx context.Context, path string, doc platform.KindS
...
+	validatedPath, err := util.ValidatePath(filepath.Join(path, readDock.Name+".yaml"))
+    if err != nil {
+        return fmt.Errorf("%s failed to validate path: %s", Emoji, err.Error())
+    }
+    
+    err = os.WriteFile(validatedPath, data, os.ModePerm)
+    if err != nil {
+        return fmt.Errorf("%s failed to write test report in yaml file. error: %s", Emoji, err.Error())
+    }
 	return nil
```
Comment (shivamsouravjha) [inline pkg/platform/yaml/testReport.go:109]: needs linter fixes

**Review by shivamsouravjha — CHANGES_REQUESTED** [review]

**Comment by shivamsouravjha** [comment]
@Ajeett01 can you please sign the CLA?

**Comment by Ajeett01** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by shivamsouravjha** [comment]
while fixing the linter issue you can add the commit with DCO signed.

use `-s` flag to do so along with your commit message.

**Comment by Ajeett01** [comment]
> while fixing the linter issue you can add the commit with DCO signed.
> 
> use `-s` flag to do so along with your commit message.

Indentation or variable name? Can you please tell the linter issue

**Comment by shivamsouravjha** [comment]
Do you have go extension installed? If so this shouldn't happen.

**Comment by Ajeett01** [comment]
> Do you have go extension installed? If so this shouldn't happen.

Yes.Please see the screenshot.
Screenshot from 2024-02-14 00-03-27

Should I commit again with little name changes. Try if it works.

**Review by shivamsouravjha — APPROVED** [review]
LGTM!

**Pull request merged**